src/bosh-dns/dns/server/handlers/factory.go

```diff
@@ -49,7 +49,7 @@ func (f *Factory) CreateForwardHandler(recursors []string, cache bool) dns.Handl
 	//
 	// The default behavior defined by DNS spec is to use
 	// "smart" recursor selection.
-	pool := NewFailoverRecursorPool(f.shuffler.Shuffle(recursors), config.SmartRecursorSelection, f.logger)
+	pool := NewFailoverRecursorPool(f.shuffler.Shuffle(recursors), config.SmartRecursorSelection, 0, f.logger)
```

Comment:
Is there a reason why you aren't using the retry config property here as well?

Reply:
Nope, I change that to use the configuration parameter sorry for that